src/__tests__/ChoiceInputField.test.tsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react'
+import {render} from '../utils/testing'
+import {render as HTMLRender, cleanup} from '@testing-library/react'
+import {axe, toHaveNoViolations} from 'jest-axe'
+import 'babel-polyfill'
+import {Checkbox, ChoiceInputField, Radio, SSRProvider} from '..'
+import {MarkGithubIcon} from '@primer/octicons-react'
+expect.extend(toHaveNoViolations)
+
+const CHECKBOXINPUTFIELD_LABEL_TEXT = 'Option one'
+const CHECKBOXINPUTFIELD_CAPTION_TEXT = 'Hint: a cool option'
+
+describe('ChoiceInputField', () => {
+  describe('snapshots', () => {
+    it('renders with a label and input', () => {
+      expect(
+        render(
+          <SSRProvider>
+            <ChoiceInputField>
+              <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+              <Checkbox />
+            </ChoiceInputField>
+          </SSRProvider>
+        )
+      ).toMatchSnapshot()
+    })
+    it('renders with a radio input', () => {
+      expect(
+        render(
+          <SSRProvider>
+            <ChoiceInputField>
+              <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+              <Radio name="textRadioInput" value="choice" />
+            </ChoiceInputField>
+          </SSRProvider>
+        )
+      ).toMatchSnapshot()
+    })
+    it('renders with a hidden label', () => {
+      expect(
+        render(
+          <SSRProvider>
+            <ChoiceInputField>
+              <ChoiceInputField.Label visuallyHidden>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+              <Checkbox />
+            </ChoiceInputField>
+          </SSRProvider>
+        )
+      ).toMatchSnapshot()
+    })
+    it('renders with a custom ID', () => {
+      expect(
+        render(
+          <SSRProvider>
+            <ChoiceInputField id="customId">
+              <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+              <Checkbox />
+            </ChoiceInputField>
+          </SSRProvider>
+        )
+      ).toMatchSnapshot()
+    })
+    it('renders as disabled', () => {
+      expect(
+        render(
+          <SSRProvider>
+            <ChoiceInputField disabled>
+              <ChoiceInputField.Label visuallyHidden>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+              <Checkbox />
+            </ChoiceInputField>
+          </SSRProvider>
+        )
+      ).toMatchSnapshot()
```

Comment:
General observation about snapshot tests: These may guard against regressions to the implementation detail, but not necessarily the actual user behaviour or the underlying html elements. I think tests like these ones should be done with testing library, which would give us greater confidence that the rendered behaviour is covered too.

Reply:
Agreed. I'll update them.